File: src/my_cpp_pkg/src/add_two_ints_client_no_oop.cpp
# include "rclcpp/rclcpp.hpp"
# include "example_interfaces/srv/add_two_ints.hpp"

using namespace std::chrono_literals;

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = rclcpp::Node::make_shared("add_two_ints_client_no_oop");

    auto client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
    while(!client->wait_for_service(1s)) {
        RCLCPP_WARN(
            node->get_logger(),
            "Interrupted while waiting for the service. Exiting.");
    }
    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 5;
    request->b = 3;
    auto future = client->async_send_request(request);
    // Wait for the result.
    rclcpp::spin_until_future_complete(node, future);
    
    auto response = future.get();
    RCLCPP_INFO(
        node->get_logger(),
        "Result of add_two_ints: %d + %d = %d",
        (int)request->a,
        (int)request->b,
        (int)response->sum);

    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/add_two_ints_client_no_oop.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


def main(args=None):
    rclpy.init(args=args)
    node = rclpy.create_node('add_two_ints_client_no_oop')

    client = node.create_client(AddTwoInts, 'add_two_ints')

    while not client.wait_for_service(timeout_sec=1.0):
        node.get_logger().info('Service not available, waiting...')

    request = AddTwoInts.Request()
    request.a = 5
    request.b = 3

    future = client.call_async(request)

    rclpy.spin_until_future_complete(node, future)

    if future.result() is not None:
        response = future.result()
        node.get_logger().info(f'Result: {request.a} + {request.b} = {response.sum}')
    else:
        node.get_logger().error('Service call failed %r' % (future.exception(),))

    node.destroy_node()
    rclpy.shutdown()